CovidShield/mobile
# Fix retrieval flow on Android

This PR fixes retrieval flow on Android. 

Resolves https://github.com/CovidShield/server/issues/70

## How to test?
- Run server locally https://github.com/CovidShield/server
- Connect mobile to local server via ngrok by changing `.env` file. 
- Change APP_ID to `com.google.android.apps.exposurenotification`.
- Make sure your device / account has EN enable. 
- Make sure `By pass signature check` is ON or `Enable app signature check` is OFF in debug mode (see screenshots below).
- Open app on Android device. 
- In `Test menu`, press `Clear exposure history and run check`. 
- You might need to set a couple of console.log in `ExposureNotificationService#performExposureStatusUpdate` to verify that the file is sent to the framework properly.

## Result

- Setup: I have 2 android devices running EN.
- The 1st device is set as positive result. When submitting the keys, I see 3 keys. I think it is collected from previous tests. The keys are submitted successfully.
```
exposureKeys [{"keyData": "CD5+GiVYuPa/FWNZUF65eA==", "rollingPeriod": 144, "rollingStartNumber": 2653776, "transmissionRiskLevel": 0}, {"keyData": "C4EvSp7ruvAtscLbi4h4Pg==", "rollingPeriod": 144, "rollingStartNumber": 2653920, "transmissionRiskLevel": 0}, {"keyData": "1mjqONFb6RLD/LxSBhEZ3w==", "rollingPeriod": 144, "rollingStartNumber": 2654064, "transmissionRiskLevel": 0}, {"keyData": "WnowdkMsDG+Dx4WqmHG3lg==", "rollingPeriod": 144, "rollingStartNumber": 2654208, "transmissionRiskLevel": 0}]
```
- The 2nd device is forced to refetch the diagnostic keys. I changed the server to return keys today too. It made 14 requests but none of them matched. Below is result of 14 requests.

Not sure what to do next to be able to test it. 🤔 

![image](https://user-images.githubusercontent.com/5274722/85086007-8b1fe700-b1a7-11ea-927b-c8fdcd7305f8.png)

## Outstanding:
- I haven't found a way to test two devices locally in test mode yet.

<img src="https://user-images.githubusercontent.com/5274722/85052150-7111e480-b166-11ea-9e0a-d64f92a698d5.png" width="240" />

<img src="https://user-images.githubusercontent.com/5274722/85052155-72dba800-b166-11ea-81bb-6e50cb0a48cf.png" width="240" />
